undertow/WFLY-17273_Global_tracking_mode_setting_in_Undertow.adoc

```diff
@@ -0,0 +1,100 @@
+= WFLY-17273 Global tracking-mode setting in Undertow
+:author:            Lin Gao
+:email:             lgao@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+There is an element: `tracking-mode` in `session-config` section of web descriptor, which is used to specify the session tracking mode, the available value can be: `COOKIE`, `URL`, `SSL`. Example in the web.xml is like:
+
+[source,xml]
+----
+<session-config>
+    <tracking-mode>COOKIE</tracking-mode>
+    <tracking-mode>URL</tracking-mode>
+</session-config>
+----
+
+When it is not defined, undertow uses `COOKIE` and falls over to `URL` if the browser does not allow cookie for the deployed web applications.
+
+When users want to have different set up for all web applications, they need to update the web.xml in each application. It would be great if undertow subsystem supports specifying the session tracking modes globally at the servlet container level.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-17273[WFLY-17273]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1980[EAP7-1980]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+WildFly Undertow subsystem
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+Provide a way to specify global session tracking modes at servlet container level. The configuration takes effect only if there are no session tracking modes specified in the deployment. So the configuration either in `web.xml` or updated in a `ServletContextListener` implementation takes precedence, which keeps consistent behavior as what it is now.
```

Comment:
Why do we need more than one tracking mode? If I understood the Jira correctly, don't we need a way to define a global tracking mode that is going to apply to all deployed applications that do not specify the session-tracking-mode in their web.xml file?

Reply:
It is possible to have multiple servlet-containers defined in undertow subsystem, so it is possible to specify different `defaullt-tracking-mode` for each servlet-container. For each servlet-container, I think it is possible to define the `defaullt-tracking-mode` to be a combination of `COOKIE`, `URL`, `SSL`(this is exclusive).

But since the default tracking modes are `COOKIE` and `URL` now already, maybe it is better to just have the `defaullt-tracking-mode` to be single choice. thanks